packages/js-dapi-client/lib/BlockHeadersProvider/BlockHeadersReader.js

```diff
@@ -33,6 +47,14 @@ class BlockHeadersReader extends EventEmitter {
    * @returns {Promise<void>}
    */
   async readHistorical(fromBlockHeight, toBlockHeight) {
+    if (this.historicalStreams.length) {
+      this.removeAllListeners(COMMANDS.HANDLE_STREAM_RETRY);
```

Comment:
Maybe we should just throw an error here so you need manually reset first and then call `readHistorical` again?

Reply:
Done